### ISSUE TITLE
What does it mean and how to fix?

### ISSUE THREAD

**Issue by sh-bd** [sarim/ibus-avro#206]
In response to my issue, Sarim Khan bhai replied something like this. I've already tried some basic stuff such as– restarting ibus and all but it doesn't help. Does anyone know— how to solve this?

![image](https://github.com/sarim/ibus-avro/assets/44128725/a6e69693-1d7e-401f-a6b2-229563d59bb4)

**Translation**: Whenever I am writing something using Avro phonetic layout and adding space at the end of the word, sometimes it automatcally goes to the beginning of the word! Do you know any solutions for this issue?
Thanks in advance.

**Comment by sh-bd** [comment]
Well, its ubuntu 22.10 64-bit (Wayland)

**Comment by sh-bd** [comment]
Well, its Ubuntu 22.10 64-bit (Wayland)

**Comment by gunnarhj** [comment]
Please let us know the output from this terminal command:

    gsettings get org.gnome.desktop.input-sources sources

**Comment by sh-bd** [comment]
Thank you @gunnarhj for your comment. It shows me– 

> [('xkb', 'us'), ('ibus', 'ibus-avro')]

> Please let us know the output from this terminal command:
> 
> ```
> gsettings get org.gnome.desktop.input-sources sources
> ```

**Comment by sarim** [comment]
It was an old problem where gtk/qt immodule was not set to ibus and ibus then runs in xim (Xserver) mode. But in recent gnomes that shouldn't be possible. @gunnarhj are you aware of similar problems in modern gnome with wayland?

@sh-bd Can you provide a screen capture video showing the issue?

**Comment by sh-bd** [comment]
Yeah, sure! Thank you again @sarim bhai for replying. Here is the screen recording–
[Screencast from 2023-12-12 20-49-26.webm](https://github.com/sarim/ibus-avro/assets/44128725/7d9768b6-9a90-46ae-96dd-5509f9f28a2f)

**Comment by gunnarhj** [comment]
@sarim wrote:
> It was an old problem where gtk/qt immodule was not set to ibus and ibus then runs in xim (Xserver) mode. But in recent gnomes that shouldn't be possible. @gunnarhj are you aware of similar problems in modern gnome with wayland?

No I'm not. GNOME sets these variables only:

```
$ env | grep -E 'XMOD|_IM'
XMODIFIERS=@im=ibus
QT_IM_MODULE=ibus
```

Gtk is handled internally by GNOME, and in Wayland it's wrong to set the `GTK_IM_MODULE` variable explicitly.

@sh-bd: To check the variables for you, can you run this command: `env | grep -E 'XMOD|_IM'`

**Comment by sh-bd** [comment]
> @sarim wrote:
> 
> > It was an old problem where gtk/qt immodule was not set to ibus and ibus then runs in xim (Xserver) mode. But in recent gnomes that shouldn't be possible. @gunnarhj are you aware of similar problems in modern gnome with wayland?
> 
> No I'm not. GNOME sets these variables only:
> 
> ```
> $ env | grep -E 'XMOD|_IM'
> XMODIFIERS=@im=ibus
> QT_IM_MODULE=ibus
> ```
> 
> Gtk is handled internally by GNOME, and in Wayland it's wrong to set the `GTK_IM_MODULE` variable explicitly.
> 
> @sh-bd: To check the variables for you, can you run this command: `env | grep -E 'XMOD|_IM'`

@gunnarhj, it shows me:
`XMODIFIERS=@im=ibus
QT_IM_MODULE=ibus
`

**Comment by gunnarhj** [comment]
@sh-bd: That's as it should be.

**Comment by sh-bd** [comment]
@gunnarhj, should I uninstall and reinstall or what?

**Comment by gunnarhj** [comment]
> should I uninstall and reinstall or what?

I don't know. I can't reproduce the issue on my machine.

**Comment by archisman-panigrahi** [comment]
@sh-bd How did you configure Avro after installing the `ibus-avro` package from APT?

**Comment by sh-bd** [comment]
@archisman-panigrahi I followed a medium tutorial that made me follow the instructions that are down below:

> 1. Open terminal and write `sudo apt-get install ibus-avro`
> 2. Run IBus (Applications -> System Tools -> IBus) from Dash
> 3. Open IBus Preferences from the top panel icon
> 4. Go to Input method
> 5. Select an input method -> Bengali -> Avro

**But here in number 5, I didn't find Avro even after restarting the PC and `ibus restart`, so I had to go the App Center and install Avro Phonetic. And after restarting this time, I found the option to select Avro as an input method. Then I followed the next instructions**

> 6. Now Click Add button to add Avro to the list
> 7. Now restart IBus from the top panel icon (Right Click -> Restart)
> 8. Now Press Ctrl+Space to toggle between English and Avro (Bengali)

Please let me know if you find anything wrong or can be modified!

**Comment by archisman-panigrahi** [comment]
@gunnarhj I have not used GNOME for several years, and cannot say if the steps followed by @sh-bd still work in recent versions of Ubuntu with GNOME.
Can you please comment on that?

**Comment by gunnarhj** [comment]
Hmm.. Those steps do not apply to Ubuntu with GNOME. Please note that IBus runs behind the scenes by default when using Ubuntu, and you normally do NOT need to bother with IBus Preferences.

These steps should be sufficient:

1. Install the `ibus-avro` package: `sudo apt install ibus-avro`
2. Log out and log in again
3. Go to Settings -> Keyboard and add Avro Phonetic to the input sources (you already did this deeming from [this comment](https://github.com/sarim/ibus-avro/issues/206#issuecomment-1852075620))

Then you can switch to Avro Phonetic via the input source indicator in the top bar, or by using the Super+Space shortcut.

**Comment by sh-bd** [comment]
@gunnarhj it doesn't help! After reinstalling, it works as it did before. How can I completely remove the previous package?

**Comment by gunnarhj** [comment]
I'm out of ideas, sorry.
